### ISSUE TITLE
Sync keeps resetting

### ISSUE THREAD

**Issue by SteveBallantine** [thomaschampagne/elevate#812]
### Step 1: Are you in the right place?

  * Your problem may have been already covered in the FAQ. Did you check https://github.com/thomaschampagne/elevate/wiki/Frequently-Asked-Questions?
  * Your problem or feature request may already exists in project issues section. Did you check https://github.com/thomaschampagne/elevate/issues?

### Step 2: Describe your environment

  * Plugin version: 6.11.0
  * Chrome/Opera/Chromium version: Chrome 73.0.3683.103
  * OS version: Windows 10.0.17134
  
### Step 3: Describe the problem:

Recently, Elevate decided it needs to resync all my activities (it seems to do this every few months and it's never been a problem before).
I followed the instructions to backup and clear the old data and then started the resync. I've got about 1600 activities so it takes a while and I just left it and went to do something else. When I came back, I saw that it stopped at 1400 activities with the message that it was limited in order to reduce the load on Strava's servers and that I should try again in a few hours.

No problem, I'll continue later.
Except when I tried to continue later, it started from 0 again. I've tried several time since then, clearing data, etc in between, but it never seems to remember what it synced in the previous session.
Now it is only syncing 400 at a time but it still starts from 0 every time.

The button in Elevate says 'Activities partially synced' so it doe seem to know it's already got some data, but it still starts again each time.

#### Steps to reproduce:

  1. Open Elevate App
  2. Sync activities
  3. Sync window says to continue sync in several hours
  4. Try to sync again and it starts from 0 activities
  
#### Actual Behavior:

Sync process does not 'remember' that it already has data and starts again from the beginning.
Below are the logs from the sync window after it has completed syncing the 400 activities. 

  ```javascript
medium.png:1 Failed to load resource: the server responded with a status of 404 ()
medium.png:1 Failed to load resource: the server responded with a status of 404 ()
jsapi?key=ABQIAAAA9S76sdh5KSPpfr65zxQHGBT-rU0CFRjazioGWeeHjJLEyYXO2RSmaFbEdGganPrQRY2l3ORmYrTPpA:23 A parser-blocking, cross site (i.e. different eTLD+1) script, https://www.google.com/uds/?file=gdata&v=1.x, is invoked via document.write. The network request for this script MAY be blocked by the browser in this or a future page load due to poor network connectivity. If blocked in this page load, it will be confirmed in a subsequent console message. See https://www.chromestatus.com/feature/5718547946799104 for more details.
google.loader.f @ jsapi?key=ABQIAAAA9S76sdh5KSPpfr65zxQHGBT-rU0CFRjazioGWeeHjJLEyYXO2RSmaFbEdGganPrQRY2l3ORmYrTPpA:23
jsapi?key=ABQIAAAA9S76sdh5KSPpfr65zxQHGBT-rU0CFRjazioGWeeHjJLEyYXO2RSmaFbEdGganPrQRY2l3ORmYrTPpA:23 A parser-blocking, cross site (i.e. different eTLD+1) script, https://www.google.com/uds/?file=gdata&v=1.x, is invoked via document.write. The network request for this script MAY be blocked by the browser in this or a future page load due to poor network connectivity. If blocked in this page load, it will be confirmed in a subsequent console message. See https://www.chromestatus.com/feature/5718547946799104 for more details.
google.loader.f @ jsapi?key=ABQIAAAA9S76sdh5KSPpfr65zxQHGBT-rU0CFRjazioGWeeHjJLEyYXO2RSmaFbEdGganPrQRY2l3ORmYrTPpA:23
jsapi?key=ABQIAAAA9S76sdh5KSPpfr65zxQHGBT-rU0CFRjazioGWeeHjJLEyYXO2RSmaFbEdGganPrQRY2l3ORmYrTPpA:23 A parser-blocking, cross site (i.e. different eTLD+1) script, https://www.google.com/uds/api/gdata/1.10/208a907929e7936990d2bf48205aa8d8/core,mediarss,geo,opensearch,atom,threading,app,gdata,finance,contacts,analytics,blogger,maps,gbase,acl,calendar,default.I.js, is invoked via document.write. The network request for this script MAY be blocked by the browser in this or a future page load due to poor network connectivity. If blocked in this page load, it will be confirmed in a subsequent console message. See https://www.chromestatus.com/feature/5718547946799104 for more details.
google.loader.f @ jsapi?key=ABQIAAAA9S76sdh5KSPpfr65zxQHGBT-rU0CFRjazioGWeeHjJLEyYXO2RSmaFbEdGganPrQRY2l3ORmYrTPpA:23
jsapi?key=ABQIAAAA9S76sdh5KSPpfr65zxQHGBT-rU0CFRjazioGWeeHjJLEyYXO2RSmaFbEdGganPrQRY2l3ORmYrTPpA:23 A parser-blocking, cross site (i.e. different eTLD+1) script, https://www.google.com/uds/api/gdata/1.10/208a907929e7936990d2bf48205aa8d8/core,mediarss,geo,opensearch,atom,threading,app,gdata,finance,contacts,analytics,blogger,maps,gbase,acl,calendar,default.I.js, is invoked via document.write. The network request for this script MAY be blocked by the browser in this or a future page load due to poor network connectivity. If blocked in this page load, it will be confirmed in a subsequent console message. See https://www.chromestatus.com/feature/5718547946799104 for more details.
google.loader.f @ jsapi?key=ABQIAAAA9S76sdh5KSPpfr65zxQHGBT-rU0CFRjazioGWeeHjJLEyYXO2RSmaFbEdGganPrQRY2l3ORmYrTPpA:23
application-fcab4c8eaeaa81596e8c9b3b7466d6807cc698acc7cbcee57f88bd3d2b07517f.js:6 Error: <svg> attribute width: A negative value is not valid. ("-32")
r @ application-fcab4c8eaeaa81596e8c9b3b7466d6807cc698acc7cbcee57f88bd3d2b07517f.js:6
application-fcab4c8eaeaa81596e8c9b3b7466d6807cc698acc7cbcee57f88bd3d2b07517f.js:6 Error: <rect> attribute width: A negative value is not valid. ("-32")
r @ application-fcab4c8eaeaa81596e8c9b3b7466d6807cc698acc7cbcee57f88bd3d2b07517f.js:6
application-fcab4c8eaeaa81596e8c9b3b7466d6807cc698acc7cbcee57f88bd3d2b07517f.js:6 Error: <svg> attribute width: A negative value is not valid. ("-32")
r @ application-fcab4c8eaeaa81596e8c9b3b7466d6807cc698acc7cbcee57f88bd3d2b07517f.js:6
application-fcab4c8eaeaa81596e8c9b3b7466d6807cc698acc7cbcee57f88bd3d2b07517f.js:6 Error: <rect> attribute width: A negative value is not valid. ("-32")
r @ application-fcab4c8eaeaa81596e8c9b3b7466d6807cc698acc7cbcee57f88bd3d2b07517f.js:6
application-fcab4c8eaeaa81596e8c9b3b7466d6807cc698acc7cbcee57f88bd3d2b07517f.js:6 Error: <svg> attribute width: A negative value is not valid. ("-32")
r @ application-fcab4c8eaeaa81596e8c9b3b7466d6807cc698acc7cbcee57f88bd3d2b07517f.js:6
application-fcab4c8eaeaa81596e8c9b3b7466d6807cc698acc7cbcee57f88bd3d2b07517f.js:6 Error: <rect> attribute width: A negative value is not valid. ("-32")
r @ application-fcab4c8eaeaa81596e8c9b3b7466d6807cc698acc7cbcee57f88bd3d2b07517f.js:6
boot.bundle.js:73 Skip handlePluginInstallOrUpgrade since we are on a sync
boot.bundle.js:73 Sync Popup. Skip handleOnFlyActivitiesSync()
VM2900:571 Uncaught TypeError: Failed to execute 'unobserve' on 'IntersectionObserver': parameter 1 is not of type 'Element'.
    at ImpressionTracker.handleDomElementRemoved (eval at e.exports (addScript.js:20), <anonymous>:571:29)
    at Mb.j (core,mediarss,geo,opensearch,atom,threading,app,gdata,finance,contacts,analytics,blogger,maps,gbase,acl,calendar,default.I.js:8)
    at ImpressionTracker.walkNodeTree (eval at e.exports (addScript.js:20), <anonymous>:489:5)
    at ImpressionTracker.Mb.j [as walkNodeTree] (core,mediarss,geo,opensearch,atom,threading,app,gdata,finance,contacts,analytics,blogger,maps,gbase,acl,calendar,default.I.js:8)
    at ImpressionTracker.walkNodeTree (eval at e.exports (addScript.js:20), <anonymous>:492:10)
    at ImpressionTracker.Mb.j [as walkNodeTree] (core,mediarss,geo,opensearch,atom,threading,app,gdata,finance,contacts,analytics,blogger,maps,gbase,acl,calendar,default.I.js:8)
    at ImpressionTracker.walkNodeTree (eval at e.exports (addScript.js:20), <anonymous>:492:10)
    at ImpressionTracker.Mb.j [as walkNodeTree] (core,mediarss,geo,opensearch,atom,threading,app,gdata,finance,contacts,analytics,blogger,maps,gbase,acl,calendar,default.I.js:8)
    at ImpressionTracker.walkNodeTree (eval at e.exports (addScript.js:20), <anonymous>:492:10)
    at ImpressionTracker.Mb.j [as walkNodeTree] (core,mediarss,geo,opensearch,atom,threading,app,gdata,finance,contacts,analytics,blogger,maps,gbase,acl,calendar,default.I.js:8)
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [1 => 1]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 1
boot.bundle.js:73 Group [1 => 1] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [2 => 2]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 2
boot.bundle.js:73 Group [2 => 2] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [3 => 3]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 3
boot.bundle.js:73 Group [3 => 3] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [4 => 4]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 4
boot.bundle.js:73 Group [4 => 4] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [5 => 5]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 5
boot.bundle.js:73 Group [5 => 5] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [6 => 6]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 6
boot.bundle.js:73 Group [6 => 6] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [7 => 7]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 7
boot.bundle.js:73 Group [7 => 7] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [8 => 8]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 8
boot.bundle.js:73 Group [8 => 8] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [9 => 9]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 9
boot.bundle.js:73 Group [9 => 9] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [10 => 10]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 10
boot.bundle.js:73 Group [10 => 10] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [11 => 11]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 11
boot.bundle.js:73 Group [11 => 11] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [12 => 12]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 12
boot.bundle.js:73 Group [12 => 12] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [13 => 13]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 13
boot.bundle.js:73 Group [13 => 13] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [14 => 14]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 14
boot.bundle.js:73 Group [14 => 14] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [15 => 15]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 15
boot.bundle.js:73 Group [15 => 15] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [16 => 16]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 16
boot.bundle.js:73 Group [16 => 16] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [17 => 17]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 17
boot.bundle.js:73 Group [17 => 17] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [18 => 18]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 18
boot.bundle.js:73 Group [18 => 18] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [19 => 19]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 19
boot.bundle.js:73 Group [19 => 19] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [20 => 20]: 20
boot.bundle.js:73 Stream length: 0, raw activities length: 20)
boot.bundle.js:73 Group handled count: 20
boot.bundle.js:73 Group [20 => 20] handled
boot.bundle.js:73 [PagesRidden] Resolving with 20 activities found
boot.bundle.js:73 Activities fetched in group [21 => 21]: 20
/activities/1222504342/streams?stream_types[]=watts_calc&stream_types[]=watts&stream_types[]=velocity_smooth&stream_types[]=time&stream_types[]=distance&stream_types[]=cadence&stream_types[]=heartrate&stream_types[]=grade_smooth&stream_types[]=altitude&stream_types[]=latlng&stream_types[]=grade_adjusted_speed:1 Failed to load resource: the server responded with a status of 429 ()
boot.bundle.js:73 Sync error Object
(anonymous) @ boot.bundle.js:73
boot.bundle.js:73 Object
(anonymous) @ boot.bundle.js:73
boot.bundle.js:73 Commited:  Object
  ```
  
#### Expected Behavior:

The sync process should remember that some activities have already been synced
  
#### Media attachement:

>You can upload/attach any images to this ticket.

#### Providing your Elevate synced athlete activities can be useful to solve the problem

> To export your activities click on "sync button" (top right and corner), then "Backup activities". You can zip the file and upload it here.

**Comment by stephan2622** [comment]
I have the same issue and I have ~4800 activities to sync.

**Comment by stephan2622** [comment]
Just want to bump this issue because it disables Elevate completely as it has no data to work with.

**Comment by keithburrows** [comment]
Hi,
I have just switched laptops and did not back up the data.  I am now having what looks like the same issue.  But mine processes 1380 activities, but when I go back to sync the rest it starts again from activity 1.
VM19:3698 GET https://www.strava.com/activities/NNNNNNNNNNN/streams?stream_types[]=watts_calc&stream_types[]=watts&stream_types[]=velocity_smooth&stream_types[]=time&stream_types[]=distance&stream_types[]=cadence&stream_types[]=heartrate&stream_types[]=grade_smooth&stream_types[]=altitude&stream_types[]=latlng&stream_types[]=grade_adjusted_speed
Sync error {streamFailure: true, activityId: NNNNNNNNNNN, statusCode: 429, statusText: "error"}

These are the key errors ^

Otherwise great App!
Ta,
Keith
